### PR TITLE
modules that call resource_action can also record before/after state

### PR DESCRIPTION
**Pull request by evgeni**
this only changes the AK module, if the design is ok, I'll update the following modules too:

* katello_content_view
* katello_repository_set
* katello_sync_plan

### CODE REVIEW

**Comment by evgeni** [comment]
test failures unrelated, fix in https://github.com/theforeman/foreman-ansible-modules/pull/568

File: plugins/modules/katello_activation_key.py [hunk 1]
```diff
@@ -251,6 +251,10 @@ def main():
             current_subscription_ids = set(item['id'] for item in current_subscriptions)
 
             if desired_subscription_ids != current_subscription_ids:
+                module.record_before('activation_keys/subscriptions', {'id': activation_key['id'], 'subscriptions': current_subscription_ids})
+                module.record_after('activation_keys/subscriptions', {'id': activation_key['id'], 'subscriptions': desired_subscription_ids})
+                module.record_after_full('activation_keys/subscriptions', {'id': activation_key['id'], 'subscriptions': desired_subscription_ids})
+
```
Comment (mdellweg) [inline plugins/modules/katello_activation_key.py]: I would have expected to see the 'after' parts after the actions, but i guess we can just assume, that it worked properly if we don't get an error.

Comment (evgeni) [reply plugins/modules/katello_activation_key.py]: yeah, that was the assumption I've been working under. can move if you prefer.

**Comment by ehelms** [comment]
Seems like folks like it, are you gonna do the other resources in this PR?

**Comment by evgeni** [comment]
@ehelms yeah, that was my plan yesterday, but then I got distracted

**Comment by evgeni** [comment]
@ehelms @mdellweg this is now ready for review

**Comment by mdellweg** [comment]
Do you want to add this functionality to the nested parameters in (domain, subnet, hostgroup, ...).

**Comment by evgeni** [comment]
You mean those which use `ensure_scoped_parameters`? That should already work as that helper uses `ensure_entity`. I can add a test to verify.

**Comment by mdellweg** [comment]
Perfect. Have i not acked this already?

**Pull request merged**